scipy/integrate/_quadpack_py.py

```diff
@@ -408,6 +440,32 @@ def quad(func, a, b, args=(), full_output=0, epsabs=1.49e-8, epsrel=1.49e-8,
     # check the limits of integration: \int_a^b, expect a < b
     flip, a, b = b < a, min(a, b), max(a, b)
 
+    if complex_func:
+        def imfunc(x, *args):
+            return np.imag(func(x, *args))
+
+        def refunc(x, *args):
+            return np.real(func(x, *args))
+
+        re_retval = quad(refunc, a, b, args, full_output, epsabs,
+                         epsrel, limit, points, weight, wvar, wopts,
+                         maxp1, limlst, complex_func=False)
+        im_retval = quad(imfunc, a, b, args, full_output, epsabs,
+                         epsrel, limit, points, weight, wvar, wopts,
+                         maxp1, limlst, complex_func=False)
+        retval = (re_retval[0] + 1j*im_retval[0],) + \
+            tuple([
+                    np.max((re_retval[i], im_retval[i])) for i in
+                    range(1, 2)])
+        msgexp = {}
+        if len(re_retval) > 2:
+            msgexp["real message"] = re_retval[2:]
+        if len(im_retval) > 2:
+            msgexp["imag message"] = im_retval[2:]
+        if len(msgexp) > 0:
+            retval = retval + (msgexp,)
```

Comment:
Could you comment on what this is supposed to do?
Are there cases in which `len(re_retval) > 2` is True and `len(im_retval) > 2` is not? I'm wondering whether these need to be separate branches or if you can just do all this when `full_output=True`?

Reply:
Ah yes! You are right.  Looking closer at the error handling, `full_output` must be true in order for the returned result to contain a message.  Otherwise, the message is presented as a warning message or it raises a `ValueError`.   That kinda renders this moot.

The purpose of this logic is to combine error messages (that are not related to an exception) in a way that users can easily distinguish which part of the integral has an issue.  I introduced this logic for a situation in which one part, either real or imaginary, raises a warning that does not occur in the other part.  The intention being to help users identify from where warning messages arise.  

Even in the `full_output=True` case, there needs to some code to protect overstepping the returned lists if one or both integrals do not return warning messages.

I'll try to implement this with fewer branches.

Reply:
Thanks. Let me know when you think it's ready for another look.

Reply:
I've replaced the three conditional statements with a single conditional on `full_output`. As we mentioned above, there is no scenario in which the real or imaginary component has more returned items.

Consequently, the "real message" and "imag message" dictionary keys don't make so much sense with `full_output`.  So, I changed these to "real output" and "imag output" respectively.

Reply:
Ok, does that mean this it's ready for another review?

Reply:
Yes.